src/http_client_async.zig

```diff
@@ -2147,6 +2255,7 @@ pub const HTTPClientResult = struct {
     fail: anyerror = error.NoError,
     redirected: bool = false,
     headers_buf: []picohttp.Header = &.{},
+    exception: ?JSC.JSValue = null,
```

Comment:
I think the exception value itself will have to be part of the code that dispatches fetch instead of living here

Reply:
I Still passing the reason using response, but is not the exception (will re write this part)